index.go

```diff
@@ -388,6 +388,11 @@ func (i *Index) createFieldIfNotExists(name string, opt FieldOptions) (*Field, e
 	return i.createField(name, opt)
 }
 
+func (i *Index) createFieldLock(name string, opt FieldOptions) (*Field, error) {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+	return i.createField(name, opt)
+}
```

Comment:
I think, you don't need this function, because we can reuse `func (i *Index) CreateField(name string, opts ...FieldOption) (*Field, error)`